src/components/AddressHistory.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import ReactLoading from 'react-loading';
+import { Link } from 'react-router-dom';
+import { TX_COUNT } from '../constants';
+import { isEqual } from 'lodash';
+import dateFormatter from '../utils/date';
+import helpers from '../utils/helpers';
+import WebSocketHandler from '../WebSocketHandler';
+import colors from '../index.scss';
+import hathorLib from '@hathor/wallet-lib';
+import PropTypes from 'prop-types';
+import PaginationURL from '../utils/pagination';
+
+
+class AddressHistory extends React.Component {
+  pagination = new PaginationURL(['hash', 'page']);
+
+  /**
+   * transactions {Array} List of transactions on the list
+   * loading {boolean} If is waiting response of data request
+   */
+  state = {
+    transactions: [],
+    loading: true,
+    firstHash: null,
+    lastHash: null,
+    hasAfter: false,
+    hasBefore: false,
+    queryParams: this.pagination.obtainQueryParams(),
+  }
+
+  componentDidMount() {
+    this.getData(this.state.queryParams);
+
+    WebSocketHandler.on('network', this.handleWebsocket);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
Maybe we should render a new `AddressHistory` instance when the parameters are changed. It would prevent this method.

Reply:
This method prevents executing unnecessary API requests, checking when it's really needed to do it. How do you think it should be done?

Reply:
You're right. Thanks!